pyproject.toml

```
@@ -85,7 +85,7 @@ include = [
     "/README.md",
     "/requirements-test.txt"
 ]
-artefacts = [
+artifacts = [
```

This was a typo, so these were not included before. Are they needed?

---

Are you SERIOUS? That's frustrating (and definitely my British fault). We certainly want these in our sdists.

---

I didn't know about that one. ([Cool graphs](https://grammarist.com/spelling/artefact-artifact/).)

We haven't coordinated British vs American spelling conventions in the codebase, but so far, it has always been without consequences. It's only a problem when a different spelling than intended goes quietly unnoticed, which isn't the case for variable and function names.

Maybe this issue should be passed up to Python itself: an unexpected word in pyproject.toml should raise an error, rather than be quietly ignored? Or maybe this specific case should be added to the codespell tool?

---

There are two options - either Hatchling could provide a json-schema, and then validate-pyproject would be able to validate the `tool.hatch` section, and/or hatchling could gain a `strict-config` setting, which I don't see present. (Scikit-build-core will have a schema in 0.5). Not sure if @ofek wants to work on it or would accept work toward it (it really depends on the backend settings implementation - it was trivial with scikit-build-core, because it was declarative dataclass-based).

---

A strict option would be doable but a schema-based implementation would be a large effort

---

I'm resolving this as I've corrected it in `main`.